src/Avalonia.X11/Screens/X11Screen.Providers.cs

```diff
@@ -211,20 +210,18 @@ private unsafe MonitorInfo[] MonitorInfos
             }
         }
 
-        public X11Screen? CreateScreenFromKey(nint key)
+        public X11Screen CreateScreenFromKey(nint key)
         {
-            var info = MonitorInfos.Where(x => x.Name == key).FirstOrDefault();
-
             var infos = MonitorInfos;
             for (var i = 0; i < infos.Length; i++)
             {
                 if (infos[i].Name == key)
                 {
-                    return new X11Screen(info, _x11, _scalingProvider, i);
+                    return new X11Screen(infos[i], _x11, _scalingProvider, i);
                 }
             }
 
-            return null;
+            return new FallBackScreen(default, _x11);
```

Comment:
I think that's not a valid behavior to return a fake screen for invalid parameter here. It's a problem of the calling code to handle a missing screen properly.

Reply:
Out of 10 `Screen` implementations, this one was the only one returning null - it's fairly recent, and should have respected the non-null contract. I guess it was missed due to the nullability mismatch here.

I've changed it to throw instead. We should never get here anyway: the key comes from the `ScreenKeys` property, and should always be found in the `for` loop above.